languages/csharp/udSDK.cs

```diff
@@ -218,24 +218,50 @@ public class udContext
         Disconnect();
     }
 
-    public void Connect(string pURL, string pApplicationName, string pKey)
+    public void ConnectWithKey(string pURL, string pKey, string pApplicationName, string appVersion, bool useDongle = false)
     {
       udError error = udError.udE_Failure;
 
-      error = udContext_TryResume(ref pContext, pURL, pApplicationName, null, 0); //Set to 1 to try use the dongle
+      error = udContext_TryResume(ref pContext, pURL, pApplicationName, null, System.Convert.ToInt32(useDongle)); //Set to 1 to try use the dongle
 
       if (error != udError.udE_Success)
-        error = udContext_ConnectWithKey(ref pContext, pURL, pApplicationName, "1.0", pKey);
+        error = udContext_ConnectWithKey(ref pContext, pURL, pApplicationName, appVersion, pKey);
 
       if (error != udError.udE_Success)
         throw new UDException(error);
+
+    }
+
+    public void ConnectInteractive(string serverURL, string applicationName, string appversion)
+    {
+
+      string approvePath = "";
+      string approveCode = "";
+      udError error = udContext_TryResume(ref pContext, serverURL, applicationName, appversion, 0); //Set to 1 to try use the dongle
+      if (error != udError.udE_Success)
+      {
+        ConnectStart(serverURL, applicationName, appversion, ref approvePath, ref approveCode);
+        Console.WriteLine("Navigate to " + approvePath + " on this device to complete udCloud login");
+        Console.WriteLine("Altenatively navigate to " + serverURL + "/link on any device and enter " + approveCode);
+        Console.WriteLine("Press any key to continue...");
+        Console.ReadKey();
+
+        try
+        {
+          ConnectComplete();
+        }
+        catch (UDException e)
+        {
+          Console.WriteLine("udCloud Login failed: " + e.Message);
+          throw e;
+        }
+      }
     }
```

Comment:
This isn't really acceptable in the core wrapper- the developing application can Console.Log if that's preferable but more likely they will need to present UI to the user or do soemthing else. We should just encourage and document with samples and text how to use Start/Complete correctly.

Reply:
In that case the user can directly call ConnectStart and ConnectComplete(). Any console application not requiring an API key will require almost exactly this method though. Obvious alternative is to move this to the sample but this is enough boilerplate to justify its own function IMO